### PR TITLE
Turn Cthulhu and JET into weakdeps

### PR DESCRIPTION
**Pull request by mkitti**
Create CthulhuExt and JETExt extension packages

Cthulhu.jl and JET.jl are rather large and complicated dependencies.

By making them weak dependencies, we increase the likelihood that SnoopCompile.jl could be loaded on development branches of Julia.

At the time of this writing, SnoopCompile.jl cannot be loaded on the Julia 1.11 branch due to an issue with Cthulhu precompilation.

### CODE REVIEW

File: src/parcel_snoopi_deep.jl [hunk 1]
```diff
@@ -1912,6 +1843,9 @@ function max_end_time(node::InferenceTimingNode, recursive::Bool=false, tmax=-on
     return tmax
 end
 
+# See ext/JETExt.jl for implementation
+function report_callees end
+
```
Comment (aviatesk) [inline src/parcel_snoopi_deep.jl:1848]:
```suggestion
```

File: test/runtests.jl [hunk 2]
```diff
@@ -1,5 +1,10 @@
 using Test
 
+if Base.VERSION >= v"1.7"
```
Comment (aviatesk) [inline test/runtests.jl:3]:
```suggestion
if VERSION >= v"1.7"
```

**Comment by aviatesk** [comment]
Thanks for your PR, @mkitti ! Unfortunately, this seems to be broken on Julia versions earlier than 1.8 (pre pkg extensions), so I'd like to go with #379 first. I'd appreciate if we can merge the change to make Cthulhu integration lazy from this PR though. Do you want to update this PR, or want me to work on that?

**Comment by mkitti** [comment]
Closing to rebase and refactor

**Pull request closed without merging**